pkg/authenticator/request/headerrequest/requestheader.go

```diff
@@ -34,6 +35,7 @@ func NewAuthenticator(providerName string, config *Config, mapper authapi.UserId
 func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.Response, bool, error) {
 	id := headerValue(req.Header, a.config.IDHeaders)
 	if len(id) == 0 {
+		audit.AddDecisionAnnotation(req, audit.DenyDecision)
```

Comment:
@stlaz, this it then wrong, right? If the basic auth comes afterwards, it will be always deny. Let me remove it.